### PR TITLE
Alerting at scale

### PR DESCRIPTION
**Pull request by wpjunior**
Based on lessons of https://landing.google.com/sre/workbook/chapters/alerting-on-slos/#alerting_at_scale

The initial motivation is to standardize a set of SLOs and help teams to classify many microservices without pain.

Could you review? @vierno, @pedrokiefer, @MayconDionisio

### CODE REVIEW

**Comment by vierno** [comment]
@wpjunior, I have a suggestion: if you could write a some inline docs and comments, it would help those reading the code.

File: README.md [hunk 1]
```diff
@@ -13,12 +13,24 @@ Easily setup a service level objective using prometheus, based on lessons from t
 
 # Generating
 
-Lookup [slo_example.yml](./slo_example.yml) to parametize SLO and generate prometheus rules running
+Lookup [slo_example.yml](./examples/slo_example.yml) to parametize SLO and generate prometheus rules running
```
Comment (vierno) [inline README.md]: 'Look the file [slo_example.yml](./examples/slo_example.yml) to see how to parametrize SLOs and generate Prometheus rules by running the following command:'

File: README.md [hunk 2]
```diff
@@ -13,12 +13,24 @@ Easily setup a service level objective using prometheus, based on lessons from t
...
 
 ```
 slo-generator -slo.path=slo_example.yml -rule.output rule.yml
 ```
 
+# SLOs at scale
+
+The Workbook suggests to create classes to simplify how to set a SLO for your services, read details about concepts [here](https://landing.google.com/sre/workbook/chapters/alerting-on-slos/#alerting_at_scale)
+
+Lookup [slo_example_with_classes.yml](./examples/slo_example_with_classes.yml) and [slo_classes.yml](./examples/slo_classes.yml) to see how to define classes and associate with your services.
```
Comment (vierno) [inline README.md]: Look at*

File: README.md [hunk 3]
```diff
@@ -13,12 +13,24 @@ Easily setup a service level objective using prometheus, based on lessons from t
...
+
+After that, you can run command specifing the classes file like this:
```
Comment (vierno) [inline README.md]: 'After that, you can run the command specifying the classes file:'

File: main.go [hunk 4]
```diff
@@ -39,15 +41,26 @@ func main() {
...
+		log.Fatal(err)
+	}
+
 	ruleGroups := &rulefmt.RuleGroups{
 		Groups: []rulefmt.RuleGroup{},
 	}
 
 	for _, slo := range spec.SLOS {
-		ruleGroups.Groups = append(ruleGroups.Groups, slo.GenerateGroupRules()...)
+		// try to use any slo class weather found
```
Comment (vierno) [inline main.go]: ~~weather~~

File: slo/classes.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package slo
+
+import "fmt"
+
+// Class represents a template of objectives
+// this is important to achieve a scalabe SLO policies
```
Comment (vierno) [inline slo/classes.go]: ~~a~~

File: slo/classes.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+type Class struct {
+	Name       string     `yaml:"name"`
+	Objectives Objectives `yaml:"objectives"`
+}
+
+type ClassesDefinition struct {
+	Classes []Class `yaml:"classes"`
+}
+
+// FindClass finds for a given name if not found return an error
```
Comment (vierno) [inline slo/classes.go]: FindClass finds for a given name, if not found return an error

**Comment by wpjunior** [comment]
Thanks, @vierno for your given review.

**Pull request merged**